From 02d24ce4001d2ef9c821566867599f41fbc3d1d1 Mon Sep 17 00:00:00 2001
From: Andreas Boll <aboll@debian.org>
Date: Sun, 16 Feb 2020 02:34:56 +0000
Subject: [PATCH 1/3] Revert "Map evdev keycode KEY_BRIGHTNESS_CYCLE to
 XF86MonBrightnessCycle"

This reverts commit ed67b29ecc639f1bafa9530cb9307fee57241458.

XF86MonBrightnessCycle currently depends on an unreleased xorgproto.

See also
https://gitlab.freedesktop.org/xkeyboard-config/xkeyboard-config/issues/123
https://gitlab.freedesktop.org/xorg/proto/xorgproto/commit/4e9f4e6136a7c35b0b3d341424eb4e0585f2f0a7

Fixes: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=921867
Fixes: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=921940
Fixes: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=922020
---
 symbols/inet | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/symbols/inet b/symbols/inet
index 049ac18..5ab6feb 100644
--- a/symbols/inet
+++ b/symbols/inet
@@ -212,7 +212,7 @@ xkb_symbols "evdev" {
     key <I246>   {      [ XF86WLAN              ]       };
 //  key <I249>   {      [ ]       }; // KEY_VIDEO_NEXT  --  drive next video source
 //  key <I250>   {      [ ]       }; // KEY_VIDEO_PREV  --  drive previous video source
-    key <I251>   {      [ XF86MonBrightnessCycle ]      }; // KEY_BRIGHTNESS_CYCLE  --  bright up, max++ == min
+//  key <I251>   {      [ ]       }; // KEY_BRIGHTNESS_CYCLE  --  bright up, max++ == min
 //  key <I252>   {      [ ]       }; // KEY_BRIGHTNESS_ZERO  --  brightness off
 //  key <I253>   {      [ ]       }; // KEY_DISPLAY_OFF  --  turn off display
     key <I254>   {      [ XF86WWAN              ]       }; // KEY_WWAN
-- 
2.7.4


From 5bb6d783f40d8aa61da2095c08a6933bb1ee7bf3 Mon Sep 17 00:00:00 2001
From: Andreas Boll <aboll@debian.org>
Date: Sun, 16 Feb 2020 02:35:03 +0000
Subject: [PATCH 2/3] Revert "Map evdev keycode KEY_KEYBOARD to XF86Keyboard"

This reverts commit d0e5e63cf59be444bec0e5b11838a9abce45226d.

This patch breaks keyboard layouts in XWayland applications when having several
keyboard layouts enabled.

See also https://bugzilla.redhat.com/show_bug.cgi?id=1587998

Fixes: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=921867
Fixes: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=921940
Fixes: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=922020

Drop this revert once we have a newer xkbcomp (>> 1.4.2) and bump
dependency on x11-xkb-utils.
---
 keycodes/evdev | 1 -
 symbols/inet   | 1 -
 2 files changed, 2 deletions(-)

diff --git a/keycodes/evdev b/keycodes/evdev
index d41799d..e08ea0a 100644
--- a/keycodes/evdev
+++ b/keycodes/evdev
@@ -285,7 +285,6 @@ default xkb_keycodes "evdev" {
 	<I255> = 255;	// #define KEY_RFKILL              247
 
 	<I372> = 372;   // #define KEY_FAVORITES           364
-	<I382> = 382;   // #define KEY_KEYBOARD            374
 	<I569> = 569;   // #define KEY_ROTATE_LOCK_TOGGLE  561
 	<I380> = 380;   // #define KEY_FULL_SCREEN         372
 
diff --git a/symbols/inet b/symbols/inet
index 5ab6feb..72cc45d 100644
--- a/symbols/inet
+++ b/symbols/inet
@@ -219,7 +219,6 @@ xkb_symbols "evdev" {
     key <I255>   {      [ XF86RFKill            ]       }; // KEY_RFKILL
 
     key <I372>  {       [ XF86Favorites          ]       };
-    key <I382>  {       [ XF86Keyboard           ]       };
     key <I569>  {       [ XF86RotationLockToggle ]       };
     key <I380>  {       [ XF86FullScreen         ]       };
 
-- 
2.7.4


From 9415d810808061cbb38832c4abf636606f3ed2ad Mon Sep 17 00:00:00 2001
From: Andreas Boll <aboll@debian.org>
Date: Sun, 16 Feb 2020 02:35:56 +0000
Subject: [PATCH 3/3] Revert "Map evdev keycode KEY_FAVORITES to XF86Favorites"

This reverts commit e6103610a2ddae633e2715c9f1da86e79b5256dd.

This patch breaks keyboard layouts in XWayland applications when having
several keyboard layouts enabled.

See also https://bugzilla.redhat.com/show_bug.cgi?id=1587998

Fixes: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=921867
Fixes: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=921940
Fixes: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=922020

Drop this revert once we have a newer xkbcomp (>> 1.4.2) and bump
dependency on x11-xkb-utils.
---
 keycodes/evdev | 1 -
 symbols/inet   | 1 -
 2 files changed, 2 deletions(-)

diff --git a/keycodes/evdev b/keycodes/evdev
index e08ea0a..8ed0fa9 100644
--- a/keycodes/evdev
+++ b/keycodes/evdev
@@ -284,7 +284,6 @@ default xkb_keycodes "evdev" {
 	<I254> = 254;	// #define KEY_WWAN                246
 	<I255> = 255;	// #define KEY_RFKILL              247
 
-	<I372> = 372;   // #define KEY_FAVORITES           364
 	<I569> = 569;   // #define KEY_ROTATE_LOCK_TOGGLE  561
 	<I380> = 380;   // #define KEY_FULL_SCREEN         372
 
diff --git a/symbols/inet b/symbols/inet
index 72cc45d..2a5ab74 100644
--- a/symbols/inet
+++ b/symbols/inet
@@ -218,7 +218,6 @@ xkb_symbols "evdev" {
     key <I254>   {      [ XF86WWAN              ]       }; // KEY_WWAN
     key <I255>   {      [ XF86RFKill            ]       }; // KEY_RFKILL
 
-    key <I372>  {       [ XF86Favorites          ]       };
     key <I569>  {       [ XF86RotationLockToggle ]       };
     key <I380>  {       [ XF86FullScreen         ]       };
 
-- 
2.7.4

